File: server/src/Makc2021.Layer1/Query/QueryResultWithOutput.cs
// Copyright (c) 2021 Maxim Kuzmin. All rights reserved. Licensed under the MIT License.

namespace Makc2021.Layer1.Query
{
    /// <summary>
    /// Результат запроса с выходными данными.
    /// </summary>
    /// <typeparam name="TOutput">Тип выходных данных.</typeparam>
    public class QueryResultWithOutput<TOutput> : QueryResult
    {
        #region Properties

        /// <summary>
        /// Выходные данные.
        /// </summary>
        public TOutput Output { get; set; }

        #endregion Properties
    }
}

File: client/makc2021-react-redux-ts/src/Layer1/Query/QueryResultWithOutput.ts
// Copyright (c) 2021 Maxim Kuzmin. All rights reserved. Licensed under the MIT License.

import { createQueryResult, QueryResult } from './QueryResult';

/**
 * Результат запроса с выходными данными.
 * @template TOutput Тип выходных данных.
 */
export interface QueryResultWithOutput<TOutput> extends QueryResult {
  /**
   * Выходные данные.
   */
  output: TOutput;
}

/**
 * Создать результат запроса.
 * @template TOutput Тип выходных данных.
 * @param queryCode Код запроса.
 * @returns Результат запроса.
 */
export function createQueryResultWithOutput<TOutput>(queryCode?: string) {
  return {
    ...createQueryResult(queryCode)
  } as QueryResultWithOutput<TOutput>;
}
